src/main/java/de/sventorben/keycloak/authentication/hidpd/AuthenticationChallenge.java

```diff
@@ -32,7 +33,9 @@ void forceChallenge() {
         String rememberMeUsername = rememberMe.getUserName();
 
         if (reauthentication.required() && context.getUser() != null) {
-            String attribute = context.getAuthenticatorConfig().getConfig().getOrDefault("userAttribute", "username");
+            String attribute = Optional.ofNullable(context.getAuthenticatorConfig())
+                .map(it -> it.getConfig().getOrDefault("userAttribute", "username").trim())
+                .orElse("username");
```

Comment:
Double check if falling back to the username attribute is correct in this case. Shouldn't it be the email as in https://github.com/sventorben/keycloak-home-idp-discovery/blob/v26.0.0/src/main/java/de/sventorben/keycloak/authentication/hidpd/discovery/email/EmailHomeIdpDiscovererConfig.java#L60